Straight_skeleton_2/include/CGAL/create_straight_skeleton_2.h

```diff
@@ -128,9 +113,12 @@ create_interior_straight_skeleton_2 ( PointIterator aOuterContour_VerticesBegin
 template<class Polygon, class K>
 boost::shared_ptr< Straight_skeleton_2<K> >
 inline
-create_interior_straight_skeleton_2 ( Polygon const& aOutContour, K const& k )
+create_interior_straight_skeleton_2 ( Polygon const& aOutContour,
+                                      K const& k,
+                                      typename std::enable_if<
+                                        ! CGAL_SS_i::has_Hole_const_iterator<Polygon>::value>::type* = nullptr)
 {
-  CGAL_precondition(aOutContour.is_simple() || !"The input polygon is not simple.");
+//  CGAL_precondition(aOutContour.is_simple() || !"The input polygon is not simple.");
```

Comment:
Yes; https://arxiv.org/pdf/1603.07401.pdf is a paper implementing some algorithm for it. 

Also it should have been CGAL::is_simple_2() or something, since we are handling models of `SequenceContainer` that might not have `is_simple()`.

I will put something clearer.